src/models/room.ts

```diff
@@ -1120,14 +1123,30 @@ export class Room extends TimelineReceipts<EmittedEvents, RoomEventHandlerMap> {
         return event;
     }
 
+    /**
+     * Get sum of threads and roon notification counts
+     * @param {String} type The type of notification count to get. default: 'total'
+     * @return {Number} The notification count.
+     */
+    public getTotalUnreadNotificationCount(type = NotificationCountType.Total): number {
+        return (this.getUnreadNotificationCount(type) ?? 0) +
+            this.getTotalThreadsUnreadNotificationCount(type);
+    }
+
+    public getTotalThreadsUnreadNotificationCount(type = NotificationCountType.Total): number {
+        return Object.keys(this.threadNotifications).reduce((total: number, threadId: string) => {
+            return total + (this.getThreadUnreadNotificationCount(threadId, type) ?? 0);
+        }, 0);
+    }
+
     /**
      * Get one of the notification counts for this room
      * @param {String} type The type of notification count to get. default: 'total'
      * @return {Number} The notification count, or undefined if there is no count
      *                  for this type.
      */
     public getUnreadNotificationCount(type = NotificationCountType.Total): number | undefined {
-        return this.notificationCounts[type];
+        return this.notificationCounts[type] ?? 0;
```

Comment:
This behaviour & signature don't match, and this is a breaking change of the contract